src/test/ui/self/arbitrary_self_types_pin_lifetime_impl_trait.stderr

```diff
@@ -1,21 +1,15 @@
-error[E0759]: `self` has an anonymous lifetime `'_` but it needs to satisfy a `'static` lifetime requirement
-  --> $DIR/arbitrary_self_types_pin_lifetime_impl_trait.rs:6:44
+error[E0700]: hidden type for `impl Trait` captures lifetime that does not appear in bounds
```

Comment:
This seems like a significant regression to diagnostics, which I'm not sure we can afford. Do we have a plan for recovering usability of the diagnostic here?

(I'm not saying that the previous diagnostic was perfect, but it did in point out the relevant pieces of code with reasonably useful explanations, and this… doesn't)

Reply:
yea, this is a problem. I can't make the span point to `self` until I do lazy TAITs, and I can't do lazy TAITs in reasonably reviewable PRs without making PRs such as this one first.

We are later actually pointing to the `&Self` here, so that's actually an improvement.

I will do a separate PR to merge before this one that will introduce the structured suggestion in nll mode and add the label with "this data with an anonymous lifetime".

Since this PR essentially removes old regionchck, I can do all improvements in nll mode and we'll see it in the .nll.stderr files.